back/app/services/xlsx_export/input_sheet_generator.rb

```diff
@@ -0,0 +1,205 @@
+          b: true,
+          alignment: { horizontal: :center, vertical: :top, wrap_text: true }
+        )
+        date_time = styles.add_style(format_code: US_DATE_TIME_FORMAT)
+        workbook.add_worksheet(name: sheetname) do |sheet|
+          sheet.add_row all_column_headers, style: column_header
+          inputs.each do |input|
+            values = all_report_field_values_for input
+            sheet.add_row do |row|
+              values.each do |value|
+                options = {}
+                if value.is_a?(ActiveSupport::TimeWithZone)
+                  options[:style] = date_time
+                end
+                row.add_cell(value, options)
+              end
+            end
+          end
+        end
+      end
+    end
+
+    private
+
+    attr_reader :inputs, :fields_in_form, :participation_method, :include_private_attributes
+
+    def supported?(field)
+      field.code != 'idea_images_attributes' # Not supported by XlsxService
+    end
+
+    def input_id_report_field
+      ComputedFieldForReport.new(column_header_for('input_id'), ->(input) { input.id })
+    end
+
+    def author_name_report_field
+      ComputedFieldForReport.new(column_header_for('author_fullname'), ->(input) { input.author_name })
+    end
+
+    def author_email_report_field
+      ComputedFieldForReport.new(column_header_for('author_email'), ->(input) { input.author.email })
+    end
+
+    def author_id_report_field
+      ComputedFieldForReport.new(column_header_for('author_id'), ->(input) { input.author_id })
+    end
+
+    def latitude_report_field
+      ComputedFieldForReport.new(column_header_for('latitude'), ->(input) { input.location_point&.coordinates&.last })
+    end
+
+    def longitude_report_field
+      ComputedFieldForReport.new(column_header_for('longitude'), ->(input) { input.location_point&.coordinates&.first })
+    end
+
+    def created_at_report_field
+      ComputedFieldForReport.new(column_header_for('created_at'), ->(input) { input.created_at })
+    end
+
+    def published_at_report_field
+      ComputedFieldForReport.new(column_header_for('published_at'), ->(input) { input.published_at })
+    end
+
+    def comments_count_report_field
+      ComputedFieldForReport.new(column_header_for('comments_count'), ->(input) { input.comments_count })
+    end
+
+    def upvotes_count_report_field
+      ComputedFieldForReport.new(column_header_for('upvotes_count'), ->(input) { input.upvotes_count })
+    end
+
+    def downvotes_count_report_field
+      ComputedFieldForReport.new(column_header_for('downvotes_count'), ->(input) { input.downvotes_count })
+    end
+
+    def baskets_count_report_field
+      ComputedFieldForReport.new(column_header_for('baskets_count'), ->(input) { input.baskets_count })
+    end
+
+    def input_url_report_field
+      ComputedFieldForReport.new(
+        column_header_for('input_url'),
+        ->(input) { Frontend::UrlService.new.model_to_url(input) }
```

Comment:
Indeed, there is no URL, because survey responses are not visible in the application, only in the export. If we add a URL, then people may link to it, while access to the page is forbidden.

Reply:
But won't this code also generate a URL for survey responses? This would mean that when admins visit the URL, they will probably get an error page?